pkg/kubelet/dockertools/manager.go

```diff
@@ -1315,3 +1334,97 @@ func (dm *DockerManager) ComputePodContainerChanges(pod *api.Pod, runningPod kub
 		ContainersToKeep:    containersToKeep,
 	}, nil
 }
+
+// Pull the image for the specified pod and container.
+func (dm *DockerManager) pullImage(pod *api.Pod, container *api.Container) error {
+	present, err := dm.IsImagePresent(container.Image)
+	if err != nil {
+		ref, err := kubecontainer.GenerateContainerRef(pod, container)
+		if err != nil {
+			glog.Errorf("Couldn't make a ref to pod %v, container %v: '%v'", pod.Name, container.Name, err)
+		}
+		if ref != nil {
+			dm.recorder.Eventf(ref, "failed", "Failed to inspect image %q: %v", container.Image, err)
+		}
+		return fmt.Errorf("failed to inspect image %q: %v", container.Image, err)
+	}
+
+	if !dm.runtimeHooks.ShouldPullImage(pod, container, present) {
+		return nil
+	}
+
+	err = dm.PullImage(container.Image)
+	dm.runtimeHooks.ReportImagePull(pod, container, err)
+	return err
+}
+
+// Sync the running pod to match the specified desired pod.
+func (dm *DockerManager) SyncPod(pod *api.Pod, runningPod kubecontainer.Pod, podStatus api.PodStatus) error {
```

Comment:
Just wanted to make sure: the code in SyncPod was moved from kubelet, so I am assuming it's safe. Is there any change that's worth paying more attention to?